src/test/java/DirectoryClientTest.java

```diff
@@ -0,0 +1,519 @@
+        GetManifestResponse getManifestResponse = directoryClient.getManifest();
+
+        // Assert
+        assertEquals("", getManifestResponse.getBody().getData().toStringUtf8());
+    }
+
+    @Test
+    void importDataTest() throws InterruptedException {
+        // Arrange
+        List<ImportElement> list = importCitadelDataList();
+        List<Object> users = list.stream()
+                .map(ImportElement::getObject)
+                .filter(object -> object != null && object.getType().equals("user"))
+                .collect(Collectors.toList());
+
+        // Act
+        directoryClient.importData(list.stream());
+
+        // Assert
+        GetObjectsResponse getObjectsResponse = directoryClient.getObjects("user");
+        assertThat(getObjectsResponse.getResultsList())
+                .usingRecursiveFieldByFieldElementComparatorOnFields("objectId_", "objectType_", "relation_", "subjectId_", "subjectType_")
+                .containsAll(users);
+
+    }
+
+    @Test
+    void exportDataTest() {
+        // Arrange & Act
+        Iterator<ExportResponse> exportedData = directoryClient.exportData(Option.OPTION_DATA);
+
+        // Assert
+        int elementCount = 0;
+        while(exportedData.hasNext()) {
+            exportedData.next();
+            elementCount++;
+        }
+
+        assertEquals(7, elementCount);
+    }
+
+    private List<ImportElement> importCitadelDataList() {
+        List<ImportElement> importElements = new ArrayList<>();
+        Object rick = Object.newBuilder()
+                .setType("user")
+                .setId("rick@the-citadel.com").build();
+        Object morty = Object.newBuilder()
+                .setType("user")
+                .setId("morty@the-citadel.com").build();
+        Object adminGroup = Object.newBuilder()
+                .setType("group")
+                .setId("admin")
+                .build();
+        Object editorGroup = Object.newBuilder()
+                .setType("group")
+                .setId("editor")
+                .build();
+        Relation rickAdminRelation = Relation.newBuilder()
+                .setObjectType("group")
+                .setObjectId("admin")
+                .setRelation("member")
+                .setSubjectType("user")
+                .setSubjectId("rick@the-citadel.com")
+                .build();
+        Relation mortyEditorRelation = Relation.newBuilder()
+                .setObjectType("group")
+                .setObjectId("editor")
+                .setRelation("member")
+                .setSubjectType("user")
+                .setSubjectId("morty@the-citadel.com")
+                .build();
+        Relation managerRelation = Relation.newBuilder()
+                .setObjectType("user")
+                .setObjectId("morty@the-citadel.com")
+                .setRelation("manager")
+                .setSubjectType("user")
+                .setSubjectId("rick@the-citadel.com")
+                .build();
```

Comment:
Is there any way that we could simplify this and not ask the consumer to client builder objects in order to create the import request?

Reply:
Added a Factory that is able to build objects, relations and object identifiers